client/service.go

```diff
@@ -107,6 +110,12 @@ type Service struct {
 	// The allowed timeout until a connection is forcefully closed.
 	DisconnectTimeout time.Duration
 
+	// The allowed timeout until a subcribe action is forcefully closed during reconnnect.
+	ResubscribeTimeout time.Duration
+
+	// Whether re-subscribe all subscriptions during reconnnect.
```

Comment:
Can you add "This can be disabled if the broker supports persistent sessions and the client is configured to request one."